pkg/client/typed/discovery/discovery_client.go

```diff
@@ -110,7 +110,7 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList
 	if err == nil {
 		apiGroup = apiVersionsToAPIGroup(v)
 	}
-	if err != nil && !errors.IsNotFound(err) && !errors.IsForbidden(err) {
```

Comment:
It's because older servers may not have /api (not found) or may be blocked (forbidden).  You'd probably have to test kubectl 1.4 against a 1.0 server to check.